modules/rhsm_manifest.py

```diff
@@ -0,0 +1,264 @@
+                sub_quantity = sum(map(lambda s: s['quantity'], subs))
+            if sub_quantity < module.params['quantity']:
+                difference = module.params['quantity'] - sub_quantity
+                attach_sub(module, manifest, difference)
+                changed = True
+        elif module.params['state'] == 'absent':
+            for sub in subs:
+                remove_sub(module, manifest, sub)
+            changed = True
+    elif module.params['state'] == 'present':
+        attach_sub(module, manifest, module.params['quantity'])
+        changed = True
+    return changed
+
+
+def remove_sub(module, manifest, sub):
+    path = "/subscription/consumers/%s/entitlements/%s" % (manifest['uuid'], sub['id'])
+    resp, info = fetch_portal(module, path, 'DELETE')
+
+
+def attach_sub(module, manifest, quantity):
+    path = "/subscription/consumers/%s/entitlements?pool=%s&quantity=%s" % (manifest['uuid'], module.params['pool_id'], quantity)
+    resp, info = fetch_portal(module, path, 'POST')
+
+
+def export_manifest(module, manifest):
+    path = "/subscription/consumers/%s/export" % (manifest['uuid'])
+    try:
+        resp, info = fetch_portal(module, path, 'GET', accept_header='application/zip')
+        f = open(module.params['path'], 'w')
+        # Read 1kb at a time to save on ram
+        while True:
+            data = resp.read(65536)  # 64K
+            if data == "":
+                break  # End of file, break while loop
+            f.write(data)
+        f.close()
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg="Failure downloading manifest, %s" % (e))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str'),
+            uuid=dict(type='str'),
+            username=dict(required=True, no_log=True),
+            password=dict(required=True, no_log=True),
+            pool_id=dict(type='str'),
+            quantity=dict(type='int'),
+            state=dict(required=True, choices=['present', 'absent']),
+            path=dict(type='path'),
+            validate_certs=dict(default=True, type='bool'),
+            portal=dict(default='https://subscription.rhn.redhat.com'),
+        ),
+        required_one_of=[['name', 'uuid']],
+        supports_check_mode=True,
+    )
+
+    name = module.params['name']
+    username = module.params['username']
+    password = module.params['password']
+
+    # Hack to add options the way fetch_url expects
+    module.params['url_username'] = username
+    module.params['url_password'] = password
+    module.params['force_basic_auth'] = True
+
+    module.params['rhsm_owner'] = get_owner(module)
+
+    manifest, man_changed = get_or_create_manifest(module)
+    if module.params['pool_id']:
+        sub_changed = get_remove_or_attach_sub(module, manifest)
+    if module.params['path']:
+        export_manifest(module, manifest)
+
+    changed = man_changed if man_changed else sub_changed
```

Comment:
I think ```change = man_changed or sub_changed``` should work just as well but you should be careful that if the module has no ```pool_id``` that ```sub_changed``` is not defined.